bundler/default/lib/saito/network.js

```diff
@@ -159,18 +169,146 @@ class Network {
         this.peers.push(peer);
         this.peers_connected++;
 
+	//
+	// initiate the handshake (verifying peers)
+	// - this is normally done in initializeWebSocket, but it is not
+	// done for remote-sockets created int he server, so we manually
+	// do it here. this adds the message emission events to the socket
+        this.app.handshake.initiateHandshake(socket);
+
+	//
+	// remote peers can do this here, as the connection is already open
+	//
+        this.app.network.requestBlockchain(peer);
+
+
         return peer;
     }
 
-    findPeer(socket) {
-        for (let i = 0; i < this.peers.length; i++) {
-            if (this.peers[i].socket_id === socket.id) {
-                return this.peers[i];
+
+
+    /**
+     * @param {string} block_hash
+     * @param {string} preferred peer (if exists); // TODO - remove duplicate function and update blockchain.js
+     */
+    async requestMissingBlock(block_hash, peer = null) { return await fetchBlock(block_hash, peer); }
+    async fetchBlock(block_hash, peer=null) {
+
+	if (peer == null) {
```

Comment:
is it required strictly to compare with `null` value? If yes, I want to suggest use strict equality operator `===`

Reply:
good idea. will push fix.

we're using operator overloading to simplify API:

  network.fetchBlock(block_hash);
  network.fetchBlock(block_hash, peer);
  
will cut "requestMissingBlock" once Sanka has pushed through the typescript update and I can wander into the blockchain file and fix that.

Reply:
if (!peer) is the way to go.

Reply:
if (!peer) is the way to go.